configure.ac

```diff
@@ -1405,11 +1412,11 @@ if test "$Py_LTO" = 'true' ; then
       case $ac_sys_system in
         Darwin*)
           # Any changes made here should be reflected in the GCC+Darwin case below
-          LTOFLAGS="-flto -Wl,-export_dynamic"
-          LTOCFLAGS="-flto"
+          LTOFLAGS="-flto$LTO_ARG -Wl,-export_dynamic"
+          LTOCFLAGS="-flto$LTO_ARG"
           ;;
         *)
-          LTOFLAGS="-flto"
+          LTOFLAGS="-flto$LTO_ARG"
```

Comment:
Are the flags (`thing` or otherwise) supported by all compilers that support LTO?

Reply:
No.  Thinlto is a more recent implementation for clang that should provide faster compile times than the default lto (full).

Reply:
https://github.com/python/cpython/blob/55a4147b16a1f96567779218f71d2ab8507ff220/configure.ac#L1362#L1460

Here's something I think about supporting the LTO options.

Reply:
Then we should probably need show an error if the compiler doesn't support it, because otherwise, it seems that is a general option that we are exposing regardless of the compiler.

Reply:
@corona10 - I like it. That's more readable and provides a warning as @pablogsal suggested.

Reply:
Nitpick: gcc doesn't have a concept of `full`, and if `--with-lto=full` is configured with gcc, no warnings are thrown and `-flto` is assumed.  Is this assumption sensible? Or would throwing an error on a nonsense input be better than assuming intent?

Reply:
I think given that these options are only available on clang we should just fail if the compiler is not clang. Adding checks to other compilers one by one is going to be suboptimal because CPython can be compiled with almost any C compiler: `xcl`, `icc`...

Reply:
https://github.com/python/cpython/pull/27231

Let's discuss the supporting option with the attached PR.
- We may check clang version
- We may check other compiler support (clang, gcc supports LTO itself, others not)